scripts/runfestat.sh

```diff
@@ -0,0 +1,220 @@
+#!/bin/bash
+
+bold=$(tput bold)
+normal=$(tput sgr0)
+unline=$(tput smul)
+
+usage() {
+
+PROGNAME=`basename $0`
+
+cat << EOFUSAGE
+
+${bold}NAME${normal}
+        ${PROGNAME} - Run FESTAT to ...
+
+${bold}USAGE${normal}
+        ${PROGNAME} -m <host-name> -x <nprocx> -y <nprocy>
+                    [ -t <threads> ]
+                    [ -b <binary-directory> ]
+                    [ -i <input-directory> ] [ -o <output-directory> ]
+                    [ -h ]
+
+${bold}DESCRIPTION${normal}
+        Script to run Forecast component of the AHNS. This script can create
+        batch submission headers and run a sample forecast.
+
+
+${bold}OPTIONS${normal}
+        -m ${unline}host name${normal}
+            The name of your platform used in logic contained in this script.
+            [ECMWF|LOCAL]
+        -x ${unline}nprocx${normal}
+            Number of processors in x-direction for 2D domain decomposition
+
+        -y ${unline}nprocy${normal}
+            Number of processors in y-direction for 2D domain decomposition
+
+        -t ${unline}threads${normal}
+            Number of OpenMP threads [ default : 1 ]
+
+        -b ${unline}binary-directory${normal}
+            PATH to FESTAT binary
+
+        -i ${unline}input-directory${normal}
+
+        -o ${unline}output-directory${normal}
+
+EOFUSAGE
+
+}
+
+set -e
+
+NPROCX=-1
+NPROCY=-1
+THREADS=1
+HOST=ECMWF                  # Host ECMWF|LOCAL
+BINDIR=$HOME/install/bin    # Where to install libraries/executables
+INPDIR=$HOME/festat_inputs  # Where to install libraries/executables
+OUTDIR=$HOME/festat_output  # Where to install libraries/executables
+UNIQTAG=$(mktemp -u XXXXXXXX)
+
+USAGE=0
+
+while getopts m:x:y:t:b:i:o:h option
+do
+  case $option in
+    m)
+       HOST=$OPTARG
+       ;;
+    x)
+       NPROCX=$OPTARG
+       ;;
+    y)
+       NPROCY=$OPTARG
+       ;;
+    t)
+       THREADS=$OPTARG
+       ;;
+    b)
+       BINDIR=$OPTARG
+       ;;
+    i)
+       INPDIR=$OPTARG
+       ;;
+    o)
+       OUTDIR=$OPTARG
+       ;;
+    h)
+       USAGE=1
+       ;;
+    *)
+       USAGE=1
+       ;;
+  esac
+done
+
+if [ ${USAGE} -eq 1 -o "$#" -eq 0 ]; then
+  usage
+  exit 1
+fi
+
+if [ ${NPROCX} -lt 1 -o ${NPROCY} -lt 1 ]; then
+  echo "Invalid NPROC settings: ${NPROCX},  ${NPROCY}"
+  exit 1
+fi
+#
+# Calculate NPROC and NPROCXY
+#
+NPROCXY=`expr ${NPROCX} \* ${NPROCY}`
+NPROC=${NPROCXY}
+
+WRK=${OUTDIR}
+echo " ... output-directory: ${WRK}"
+
+WDIR=wrkFestat_${HOST}_${NPROCX}_${NPROCY}_${THREADS}
+d_FESTAT=${WRK}/${WDIR}
+echo " ... run-directory   : ${d_FESTAT}"
+mkdir -p ${d_FESTAT}
+cd ${d_FESTAT}
+
+if [ ${HOST} == "ECMWF" ] ; then
+
+  cat << EOFBATCH > batch.tmp
+#SBATCH --cpus-per-task=8
+#SBATCH --error=festat.log
+#SBATCH --job-name=FESTAT
+#SBATCH --nodes=37
+#SBATCH --output=festat.log
+#SBATCH --qos=np
+#SBATCH --ntasks-per-node=27
+#SBATCH --time=02:00:00
+#
+
+EXP=SURF_CARRA2_CY46
```

Comment:
I do not believe we should have a dependency on a Harmonie experiement to run this.